eng/templates/Azure.ResourceManager.Template/src/Properties/AssemblyInfo.cs

```diff
@@ -5,4 +5,5 @@
 
 [assembly: Azure.Core.AzureResourceProviderNamespace("ProviderFullName")]
 
+[assembly: InternalsVisibleTo("DynamicProxyGenAssembly2, PublicKey=0024000004800000940000000602000000240000525341310004000001000100c547cac37abd99c8db225ef2f6c8a3602f3b3606cc9891605d02baa56104f4cfc0734aa39b93bf7852f7d9266654753cc297e7d2edfe0bac1cdcf9f717241550e0a7b191195b7667bb4f64bcb8e2121380fd1d9d46ad2d92d2d15605093924cceaf74c4861eff62abf69b9291ed0a340e113be11e6a7d3113e92484cf7045cc7")]
```

Comment:
Why do we need this?

Reply:
That's because we use some private cache methods in mgmt plane to improve the performance and the process of creating proxy will fail without this line, there is also a [note](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/core/Azure.Core.TestFramework#note-on-privatenon-virtual-fields-in-your-clients-such-as-_clientdiagnostics-and-internalsvisibleto) in the readme of test framework.